noxfile.py

```diff
@@ -28,7 +30,28 @@
 MD_PATHS = ['*.md']
 
 
-def run_readable(session, mode="fmt"):
+@functools.lru_cache
+def _list_files() -> list[Path]:
+    file_list = []
+    for cmd in (
+        ["git", "ls-files"],
+        ["git", "ls-files", "--others", "--exclude-standard"],
+    ):
+        cmd_result = subprocess.run(cmd, check=True, text=True, capture_output=True)
+        file_list.extend(cmd_result.stdout.splitlines())
+    file_paths = [Path(p) for p in file_list]
+    return file_paths
+
+
+def list_files(suffix: str | None = None) -> list[Path]:
+    """List all non-files not-ignored by git."""
+    file_paths = _list_files()
+    if suffix is not None:
```

Comment:
how about passing the `suffix` into `_list_files()` and deal with it there, since you're already iterating through the list there?

Reply:
that wouldn't allow to leverage the cache if start calling this multiple times for different kinds of files

for example, it would be nice to use it in `readable` formatter, since its wildcard support is also subpar, but I opted to skip that in this PR just to cut some corners and save time.